### DISCUSSION TITLE
Facing Issue while trying to Hit Custom API in Airbyte

### DISCUSSION THREAD

**Discussion by sekhardunna906** [airbytehq/airbyte#34557] [General]
Hi @everyone,

Need Help!!!

I have an custom API which i am trying to call in Airbyte. I am facing an issue like **ERROR
Server temporarily unavailable (http.502.iDJnxkBNLVTvQNYMEJF1QR)** .  I checked my logs, everything looks fine but still i am not getting an response in Airbyte and the 2 streams(**sprint_details**, **items_details**) response is coming perfectly in Airbyte.

The response for **timesheet_details** is more, maybe for that reason in Airbyte it's causing an error. I need solution for that.

I am attaching the screenshot of Airbyte Tab for the reference.
<img width="959" alt="error" src="https://github.com/airbytehq/airbyte/assets/157820480/8dbaf77f-a673-4be2-9ace-0a63158b28da">

This is my log file screenshot.
<img width="794" alt="image" src="https://github.com/airbytehq/airbyte/assets/157820480/7371e4f9-1016-406b-9b05-4067efc2034b">

Help from your side will be highly appreciable.
Thanks

**Comment by sajarin** [comment]
From the information you've provided, it seems like you're encountering a 502 error when trying to connect to a custom API using Airbyte. This error typically indicates that a server, while acting as a gateway or proxy, received an invalid response from an inbound server it accessed while attempting to fulfill the request.

One possible reason for this could be that the server is taking too long to respond due to the large response for timesheet_details, causing a timeout. You might want to check the server timeout time and consider increasing it if necessary.

Additionally, to help us understand better if the error is related to the platform or the specific API you're trying to connect to, could you try connecting to other sources like Github, CSV, or a local Postgres database and see if you encounter the same issue?

[a comment by an automated account is not shown]
